pkg/auth/providers/saml/saml_provider.go

```diff
@@ -156,12 +156,13 @@ func PerformSamlLogoutAll(apiContext *types.APIContext, token *v3.Token) error {
 	if err != nil {
 		return err
 	}
-	if userAttributes == nil {
-		logrus.Debugf("SAML [logout-all]: UserAttributes for rancher user %q not found", userName)
-		return fmt.Errorf("SAML [logout-all]: UserAttributes for rancher user %q not found", userName)
-	}
 
-	userAtProvider := userAttributes.ExtraByProvider[providerName]["username"][0]
+	// Incremental extraction of the user at provider info
```

Comment:
nit: We probably don't need this comment as it doesn't add much to the code that follows

Reply:
Done: 52ae9b76f00121be2f3e5756feb8eac7ac0734c0